addons/web/controllers/binary.py

```diff
@@ -126,22 +126,27 @@ def content_assets(self, filename=None, unique=ANY_UNIQUE, nocache=False, assets
                     bundle_name, rtl, asset_type, autoprefix = rw_env['ir.asset']._parse_bundle_name(filename, debug_assets)
                     css = asset_type == 'css'
                     js = asset_type == 'js'
+                    binary = asset_type == 'binary'
+                    extension = filename.rsplit('.', 1)[1]
```

Comment:
```suggestion
                    extension = os.path.splitext(filename)[1]
```

`os.path.splitext` correctly works with hidden files, split does not
but note that it keeps the leading dot, so if you don't want it:

```suggestion
                    extension = os.path.splitext(filename)[1].removeprefix('.')
```